internal/db/db_sqlite.go

```diff
@@ -113,7 +113,10 @@ func (s *Sqlite) CreateFile(ctx context.Context, file *snips.File, maxFileCount
 		return ErrFileLimit
 	}
 
-	file.ID = id.New()
+	if file.ID == "" {
+		file.ID = id.New()
+	}
```

Comment:
I'd prefer if we added a new field separate from the file ID.

Allowing any text could get tricky especially when invoking the SSH commands. For example, if someone renamed to `@my!$FILE` it wouldn't play well with the ssh command to get the file:

```
ssh f:@my!$FILE@snips.sh
```

For the UI/list view, we can overwrite the rendered rows to show the name instead of the id if the name is non-nil.